class/voyage.class.php

```diff
@@ -240,14 +215,131 @@ public function __construct(DoliDB $db)
 	 */
 	public function create(User $user, $notrigger = false)
 	{
+        global $conf;
+
+//        if (!empty(GETPOST('idProduct','int'))){
+//            $idProduct = GETPOST('idProduct','int');
+//            $this->array_options['options_product'] = $idProduct;
+//        }
+
+        if(empty($this->tarif) && !empty($this->array_options['options_tag']))
+        {
+            $TTag = explode(',', $this->array_options['options_tag']);
+            $tarift = $this->findTarifWithOneTag($TTag[0]);
+
+            foreach ($TTag as $valueTag)
+            {
+                if ($tarift > $this->findTarifWithOneTag($valueTag))
+                {
+                    $tarift = $this->findTarifWithOneTag($valueTag);
+                }
+            }
+            $this->tarif = $tarift;
+        }
+        else if(empty($this->tarif) && empty($this->array_options['options_tag'])){
+            $this->tarif = $conf->global->VOYAGEBUILDER_MYPARAM1;
+        }
```

Comment:
Au lieu de mettre plusieurs lignes de codes, il faudrait ajouter une seule ligne "$this->setTarifByTag" et développer par la suite la fonction "setTarifByTag"